pkg/getter/httpgetter.go

```diff
@@ -34,7 +35,15 @@ type httpGetter struct {
 func (g *httpGetter) Get(href string) (*bytes.Buffer, error) {
 	buf := bytes.NewBuffer(nil)
 
-	resp, err := g.client.Get(href)
+	// Set a helm specific user agent so that a repo server and metrics can
+	// separate helm calls from other tools interacting with repos.
+	req, err := http.NewRequest("GET", href, nil)
```

Comment:
how come we're stripping out `g.client` here and calling `http.NewRequest` directly? Should we introduce `g.client.NewRequest` if the existing `httpGetter` struct does not implement that?